### PR TITLE
Added ray version of the html2parquet transform

### PR DESCRIPTION
**Pull request by sungeunan-ibm**
## Why are these changes needed?
Added ray version of the html2parquet transform

## Related issue number (if any).
Issue #635

### CODE REVIEW

**Comment by shahrokhDaijavad** [comment]
Great. Thank you very much @sungeunan-ibm for finishing this. It looks like the conflict in yml is the result of 1) The move of this transform from universal to the language folder and 2) The fact that a new hap transform was added yesterday. But, the main reviewer for the correct implementation of ray is @daw3rd.

**Comment by sungeunan-ibm** [comment]
Thank you @shahrokhDaijavad! @daw3rd Could you advise on how I should resolve these conflicts? Should I simply regenerate the file by running the following commands?
```
cd .github/workflows
rm test-language-html2parquet.yml 
make
```

File: .github/workflows/test-language-html2parquet.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #
 # DO NOT EDIT THIS FILE: it is generated from test-transform.template,  Edit there and run make to change these files
 #
+<<<<<<<< HEAD:.github/workflows/test-language-html2parquet.yml
```
Comment (daw3rd) [inline .github/workflows/test-language-html2parquet.yml]: How did you check this in?  I thought git would not allow it.  Anyway.  I recommend the following.
1. merge the latest dev branch into your branch
2. run `make` in this directory and you should get a proper version of this file.

File: .github/workflows/test-universal-hap.yml [hunk 2]
```diff
@@ -1,7 +1,11 @@
 #
 # DO NOT EDIT THIS FILE: it is generated from test-transform.template,  Edit there and run make to change these files
 #
+<<<<<<<< HEAD:.github/workflows/test-language-html2parquet.yml
```
Comment (daw3rd) [inline .github/workflows/test-universal-hap.yml]: Same here

File: transforms/language/html2parquet/ray/Makefile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+run-local-sample: .transforms.run-local-sample
+
+run-local-python-sample: .transforms.run-local-python-sample
+
+#run-s3-ray-sample: .transforms.run-s3-ray-sample
+
+minio-start:	.minio-start
+
+kind-load-image:: .transforms.kind-load-image
```
Comment (daw3rd) [inline transforms/language/html2parquet/ray/Makefile:57]: Please update the following rules to match those in other transforms from the latest dev branch.  For example, https://github.com/IBM/data-prep-kit/blob/eadce898542f00493b7c0cc6cdde04d86dc7be6b/transforms/universal/ededup/Makefile#L56

File: transforms/language/html2parquet/ray/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+This module implements the ray version of the [html2parquet transform].
+
+The HTML conversion is using the [Trafilatura](https://trafilatura.readthedocs.io/en/latest/usage-python.html).
+
+## Prometheus metrics
+
+The transform will produce the following statsd metrics:
+
+| metric name                      | Description                                                      |
```
Comment (daw3rd) [inline transforms/language/html2parquet/ray/README.md:11]: These are not part of the output parquet file.  I think you need to tell the reader where these can be viewed.   Also, please link from here to the python readme for annotations and configuration.

**Review by daw3rd — DISMISSED** [review]

File: transforms/language/html2parquet/ray/Makefile (hunk 3 above)
Comment (sungeunan-ibm) [reply transforms/language/html2parquet/ray/Makefile:57]: The Makefile you are referring to is located in the ray folder, while the example provided is directly under the ededup folder. I checked the Makefile for the ededup transform within the ray folder, and it matches the same structure.

File: transforms/language/html2parquet/ray/README.md (hunk 4 above)
Comment (sungeunan-ibm) [reply transforms/language/html2parquet/ray/README.md:11]: Actually, I'm not entirely sure where these metrics can be viewed. In the Ray implementation, I borrowed the pdf2parquet code, but I’m not very familiar with Prometheus metrics. These metrics are generated from the `Html2ParquetRayTransform` class, and they can be viewed through logging. After some investigation, it appears that you can access these metrics in the Prometheus dashboard under the relevant namespace for your application. However, my experience with this is limited. I would greatly appreciate your assistance in adding more detailed descriptions!

File: transforms/language/html2parquet/ray/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+[project]
+name = "dpk_html2parquet_transform_python"
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/pyproject.toml]:
```suggestion
name = "dpk_html2parquet_transform_ray"
```

File: transforms/language/html2parquet/ray/Makefile [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+# transforms should be able to reuse.  However, feel free
+# to override/redefine the rules below. 
+
+# $(REPOROOT)/.make.versions file contains the versions
+
+TRANSFORM_NAME= html2parquet
+
+include $(REPOROOT)/transforms/.make.transforms
+
+venv::	.transforms.python-venv
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
venv::	.transforms.ray-venv
```

File: transforms/language/html2parquet/ray/Makefile [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+test::	.transforms.python-test
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
test::	.transforms.ray-test
```

File: transforms/language/html2parquet/ray/Makefile [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+clean:: .transforms.clean
+
+image:: .transforms.python-image
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
image:: .transforms.ray-image
```

File: transforms/language/html2parquet/ray/Makefile [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+test-src:: .transforms.test-src
+
+setup:: .transforms.setup
+
+build:: build-dist image
+
+publish: publish-image
+
+publish-image:: .transforms.publish-image-python
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
publish-image:: .transforms.publish-image-ray
```

File: transforms/language/html2parquet/ray/Makefile [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+# distribution versions is the same as image version.
+set-versions:
+	$(MAKE) TRANSFORM_PYTHON_VERSION=$(HTML2PARQUET_PYTHON_VERSION) TOML_VERSION=$(HTML2PARQUET_PYTHON_VERSION) .transforms.set-versions 
+        
+build-dist:: set-versions .defaults.build-dist 
+
+publish-dist:: .defaults.publish-dist
+
+test-image:: .transforms.python-test-image
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
test-image:: .transforms.ray-test-image
```

File: transforms/language/html2parquet/ray/Makefile [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+run-cli-sample:
+	$(MAKE) RUN_FILE=$(TRANSFORM_NAME)_transform_python.py \
+	    RUN_ARGS="	\
+	    --data_local_config \" { 'input_folder' : '../test-data/input', 'output_folder' : '../output' } \"  \
+	    --data_files_to_use \"['.html','.zip']\"  
+
+run-local-sample: .transforms.run-local-sample
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
run-local-sample: .transforms.run-local-ray-sample
```

File: transforms/language/html2parquet/ray/Makefile [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+run-local-python-sample: .transforms.run-local-python-sample
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
```

**Review by touma-I — CHANGES_REQUESTED** [review]

File: .github/workflows/test-code-license_select-kfp.yml [hunk 13]
```diff
@@ -102,6 +102,7 @@ jobs:
             - name: Test KFP libs (shared and v2) and run a workflow
               timeout-minutes: 120
               run: |
+
```
Comment (touma-I) [inline .github/workflows/test-code-license_select-kfp.yml]: Why is this file being changed by this PR?

File: transforms/language/html2parquet/Makefile [hunk 14]
```diff
@@ -53,7 +53,7 @@ docker-save-image::
 	@# Help: Recursively make $@ in all subdirs 
 	$(MAKE) RULE=$@ .recurse
 
-.PHONY: workflow-venv
+.PHONY: workflow-vent
```
Comment (touma-I) [inline transforms/language/html2parquet/Makefile:56]: There is no kfp support yet. Why is this in here ?

File: transforms/language/html2parquet/ray/Makefile [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+# Include a library of common .transform.* targets which most
+# transforms should be able to reuse.  However, feel free
+# to override/redefine the rules below. 
+
+# $(REPOROOT)/.make.versions file contains the versions
+
+TRANSFORM_NAME= html2parquet
+
+include $(REPOROOT)/transforms/.make.transforms
+
```
Comment (touma-I) [inline transforms/language/html2parquet/ray/Makefile]:
```suggestion
BASE_IMAGE=${RAY_BASE_IMAGE}
```

**Review by touma-I — APPROVED** [review]

**Pull request merged**